File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1
        max_area = 0

        while l<r:
            curr_area = min(height[l], height[r]) * (r-l)
            if height[l] < height[r]: l += 1
            else: r -= 1

            max_area = max(max_area, curr_area)
        
        return max_area

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size()-1;
        int maxArea = INT_MIN;

        while(l<r){
            int currArea = min(height[l], height[r])*(r-l);
            maxArea = max(maxArea, currArea);
            if(height[l] < height[r]) l++;
            else r--;
        }
        return maxArea;
    }
};